command/agent/consul/structs.go

```diff
@@ -21,7 +21,7 @@ func BuildAllocServices(
 			AllocID: alloc.ID,
 			Group:   alloc.TaskGroup,
 		},
-		Services: taskenv.InterpolateServices(taskenv.NewBuilder(mock.Node(), alloc, nil, alloc.Job.Region).Build(), tg.Services),
+		Services: taskenv.NewBuilder(mock.Node(), alloc, nil, alloc.Job.Region).Build().InterpolateServices(tg.Services),
```

Comment:
We have the `TaskGroup` in scope here, so shouldn't we be calling `tg.InterpolatedServices(taskenv.NewBuilder(...).Build())`?

(Also, I realize this is take from the original code but why do we call `mock.Node()` here rather than passing in the `node` parameter?)